server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -71,26 +71,35 @@
     public static final String NOOP_PIPELINE_NAME = "_none";
 
     private final ClusterService clusterService;
-    private final PipelineStore pipelineStore;
-    private final PipelineExecutionService pipelineExecutionService;
+    private final Map<String, Processor.Factory> processorFactories;
+
+    // Ideally this should be in IngestMetadata class, but we don't have the processor factories around there.
+    // We know of all the processor factories when a node with all its plugin have been initialized. Also some
+    // processor factories rely on other node services. Custom metadata is statically registered when classes
+    // are loaded, so in the cluster state we just save the pipeline config and here we keep the actual pipelines around.
+    private volatile Map<String, Pipeline> pipelines = new HashMap<>();
+
+    private final ThreadPool threadPool;
+
+    private final StatsHolder totalStats = new StatsHolder();
+
```

Comment:
nit: remove these white lines?

Reply:
yea, that would be nicer => cleaning it up :)